src/udp.cpp

```diff
@@ -237,7 +237,18 @@ namespace oxen::quic
                 ev_,
                 sock_,
                 EV_READ | EV_PERSIST,
-                [](evutil_socket_t, short, void* self) { static_cast<UDPSocket*>(self)->receive(); },
+                [](evutil_socket_t, short, void* self) {
+#ifndef NDEBUG
+                    check_rv(
+#endif
+                            static_cast<UDPSocket*>(self)
+                                    ->receive()
+#ifndef NDEBUG
+                                    .error_code,
+                            "udp::receive()")
+#endif
```

Comment:
I don't think this should be `check_rv`'ed for debug builds: it will throw upon failure, which makes the debug and non-debug paths through here substantially different.

Reply:
Checking the return value and logging is fine, just not via `check_rv` because of that throwing behaviour.